lib/internal/http2/compat.js

```diff
@@ -4,9 +4,8 @@ const { Object, ObjectPrototype, Reflect } = primordials;
 
 const assert = require('internal/assert');
 const Stream = require('stream');
-const Readable = Stream.Readable;
-const binding = internalBinding('http2');
-const constants = binding.constants;
+const { Readable } = Stream;
+const { constants } = internalBinding('http2');
```

Comment:
Nit: why not using the pattern used for `codes` below?

```js
const {
  constants: {
    HTTP2_HEADER_AUTHORITY,
    // ....
  }
} = internalBinding('http2');
```

Reply:
I can do that, although the case is slightly different:

it's not only `codes` being destructured, but also `hideStackFrames`. what I see people usually tend to do is  not to use deeper destructuring when it's only off of one property, meaning:

```js
// do:
const { a } = b.c
// don't do:
const { c: { a } } = b
// but do:
// that would be ok, because we have 2 props to destructure off, c and d:
const { c: { a }, d } = b
```

the reason I guess is besides readability, it would also a few less characters.

that being said, I can certainly pull up all constants to the require section, and if needed, do deeper destructuring as well. I just didn't wanna cause too much code churn and have the PR being turned down. in any of my own projects I would have done it as well ...

Reply:
@lpinca  I don't have a strong opinion on that topic though, in fact, I did it as well quite a lot in my earlier destructuring days. also seems to be consistent with `codes` in this particular file.

-> I pulled up the constants to the require call and applied deep destructuring.